lib/route-recognizer.js

```diff
@@ -89,13 +89,13 @@ EpsilonSegment.prototype = {
   generate: function() { return ""; }
 };
 
-function parse(route, names, specificity) {
+function parse(allSegments, route, names, specificity) {
   // normalize route as not starting with a "/". Recognition will
   // also normalize.
   if (route.charAt(0) === "/") { route = route.substr(1); }
 
   var segments = route.split("/");
-  var results = new Array(segments.length);
+  //var results = new Array(segments.length);
```

Comment:
You need to look at the whole change, these arrays were just concat with each other to allSegments, so one growing array is faster than several preallocated arrays that are concatenated